tests/test_json_schema.py

```diff
@@ -0,0 +1,133 @@
+import json
+import os
+import tempfile
+import textwrap
+from typing import Type
+
+import pytest
+from jsonschema import ValidationError
+
+from tests.yahp_fixtures import ChoiceHparamParent, PrimitiveHparam, ShavingBearsHparam
+from yahp.hparams import Hparams
+
+
+@pytest.mark.parametrize('hparam_class,success,data', [
+    [
+        PrimitiveHparam, True,
+        textwrap.dedent("""
+            ---
+            intfield: 1
+            strfield: hello
+            floatfield: 0.5
+            boolfield: true
+            enumintfield: ONE
+            enumstringfield: pytorch_lightning
```

Comment:
nit: let's not reference competitors in the codebase except for specific functionality

```suggestion
            enumstringfield: world
```